pubpub/pubpub
# feature: Text alignment in Prosemirror content

Since I am migrating the Layout editor away from `FormattingBarLegacy`, and the rich text Layout block editor has its own sort of hacked-in text alignment option, I thought it would be neat to try to support text alignment across all PubPub content. So, here it is. Paragraphs, headers, and lists can all be centered. The document body, blockquotes, and table cells all act as containers whose immediate children may be aligned — so we don't end up with contradicting alignment rules between parent nodes and their children.

_Screenshot:_
![image](https://user-images.githubusercontent.com/2208769/112059647-799ead80-8b32-11eb-9239-0a94b2cff33f.png)

_Test plan:_
- Ensure that the following elements can be aligned and re-aligned: paragraphs, headings, lists
- Ensure that table cells and table heading cells have their own alignment separate from that of the document
- Ensure that the alignment buttons in the formatting bar work properly: if all of the currently selected content shares an alignment, that alignment is highlighted
- Ensure that alignment is carried through copy+paste operations if you select an entire paragraph or other alignable node (this will not work if your selection is entirely contained within the paragraph)
- Ensure that alignment works in PDF and HTML exports
